node/src/store/header_ranges.rs

```diff
@@ -1,126 +1,151 @@
-use std::fmt::Display;
-use std::ops::RangeInclusive;
+use std::borrow::Borrow;
+use std::fmt::{self, Debug, Display};
+use std::iter;
+use std::mem;
+use std::ops::{RangeBounds, RangeInclusive, Sub};
 use std::vec;
 
-#[cfg(any(test, feature = "test-utils"))]
-use celestia_types::test_utils::ExtendedHeaderGenerator;
 use celestia_types::ExtendedHeader;
 use serde::{Deserialize, Serialize};
 use smallvec::SmallVec;
 
 use crate::store::utils::{ranges_intersection, try_consolidate_ranges, RangeScanResult};
 use crate::store::StoreError;
 
-pub type HeaderRange = RangeInclusive<u64>;
+pub type BlockRange = RangeInclusive<u64>;
```

Comment:
just throwing ideas at the wall, but maybe `HeightRanges`?

Reply:
I think `BlockRanges` feels nicer.